docs/spark.rst

```diff
@@ -0,0 +1,138 @@
+MPI that makes running Horovod jobs in Spark clusters easy.
+
+In situations where training data originates from Spark, this enables
+a tight model design loop in which data processing, model training, and
+model evaluation are all done in Spark.
+
+A toy example of running a Horovod job in Spark is provided below:
+
+.. code-block:: bash
+
+    $ pyspark
+    [PySpark welcome message]
+
+    >>> def fn(magic_number):
+    ...   import horovod.torch as hvd
+    ...   hvd.init()
+    ...   print('Hello, rank = %d, local_rank = %d, size = %d, local_size = %d, magic_number = %d' % (hvd.rank(), hvd.local_rank(), hvd.size(), hvd.local_size(), magic_number))
+    ...   return hvd.rank()
+    ...
+    >>> import horovod.spark
+    >>> horovod.spark.run(fn, args=(42,))
+    Running 16 processes...
+    [Stage 0:>                                                        (0 + 16) / 16]
+    Hello, rank = 15, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 13, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 8, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 9, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 10, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 11, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 6, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 4, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 0, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 1, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 2, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 5, local_rank = 1, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 3, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 12, local_rank = 0, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 7, local_rank = 3, size = 16, local_size = 4, magic_number = 42
+    Hello, rank = 14, local_rank = 2, size = 16, local_size = 4, magic_number = 42
+    [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
+    >>>
+
+**End-to-end example**
+
+`keras_spark_rossmann.py script <../examples/keras_spark_rossmann.py>`__ provides
+an example of end-to-end data preparation and training of a model for the
+`Rossmann Store Sales <https://www.kaggle.com/c/rossmann-store-sales>`__ Kaggle
+competition.
+
+It is inspired by an article `An Introduction to Deep Learning for Tabular Data <https://www.fast.ai/2018/04/29/categorical-embeddings/>`__
+and leverages the code of the notebook referenced in the article.
+
+The example is split into three parts:
+1. The first part performs complicated data preprocessing over an initial set
+of CSV files provided by the competition and gathered by the community.
```

Comment:
I think for an ordered list the sentence must all be on 1 line

Reply:
Good catch! Fixed it.